store/models/v1.go

```diff
@@ -0,0 +1,98 @@
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/utils"
+	"go.uber.org/multierr"
+	null "gopkg.in/guregu/null.v3"
+)
+
+type AssignmentSpec struct {
+	Assignment Assignment `json:"assignment"`
+	Schedule   Schedule   `json:"schedule"`
+}
+
+type Assignment struct {
+	Subtasks []Subtask `json:"subtasks"`
+}
+
+type Subtask struct {
+	Type   string `json:"adapterType"`
+	Params JSON   `json:"adapterParams"`
+}
+
+type Schedule struct {
+	EndAt       Time        `json:"endAt"`
+	Hour        null.String `json:"hour"`
+	Minute      null.String `json:"minute"`
+	DayOfMonth  null.String `json:"dayOfMonth"`
+	MonthOfYear null.String `json:"monthOfYear"`
+	DayOfWeek   null.String `json:"dayOfWeek"`
+	RunAt       []Time      `json:"runAt"`
+}
+
+func (s Schedule) hasCron() bool {
+	return s.Minute.Valid || s.Hour.Valid || s.DayOfMonth.Valid ||
+		s.MonthOfYear.Valid || s.DayOfWeek.Valid
+}
+
+func (s Schedule) toCron() Cron {
+	return Cron(fmt.Sprintf("0 %v %v %v %v %v",
+		cronUnitOrDefault(s.Minute),
+		cronUnitOrDefault(s.Hour),
+		cronUnitOrDefault(s.DayOfMonth),
+		cronUnitOrDefault(s.MonthOfYear),
+		cronUnitOrDefault(s.DayOfWeek),
+	))
+}
+
+func cronUnitOrDefault(s null.String) string {
+	if s.Valid {
+		return s.String
+	}
+	return "*"
+}
+
+func appendCronInitiator(initiators []Initiator, s AssignmentSpec) []Initiator {
+	if s.Schedule.hasCron() {
+		initiators = append(initiators, Initiator{
+			Type:     "cron",
+			Schedule: s.Schedule.toCron(),
```

Comment:
It's interesting that these private methods are used by another type, `AssignmentSpec`, making them effectively `internal` to the package. I love the coverage we have now but would have personally went with making a public method `schedule.ToCron()` or even `schedule.ToInitiator()` and tested that rather than having the verbose assignment spec below.

With go having multiple types in a file, I'm for having public methods on types when they're used by other types even if they're only used inside the same package. It makes testing easier and stops us from making a super `Main` method that relies on a lot of privates.